plugins/woocommerce/src/Admin/API/Notes.php

```diff
@@ -39,9 +39,22 @@ class Notes extends \WC_REST_CRUD_Controller {
 	 *
 	 * @var array
 	 */
-	protected $allowed_promo_notes = array(
-		'wcpay-promo-2022-us-incentive-20-off',
-	);
+	protected $allowed_promo_notes;
+
+	/**
+	 * Initialize allowed promo notes.
+	 */
+	public function __construct() {
+		/**
+		 * Filter the allowed promo notes.
+		 *
+		 * Enables adding allowed promo notes for experimental-activate-promo.
+		 *
+		 * @param array     $promo_notes    Array of allowed promo notes.
+		 * @since 7.8.0
+		 */
+		$this->allowed_promo_notes = apply_filters( 'woocommerce_admin_allowed_promo_notes', $this->allowed_promo_notes );
```

Comment:
Do we have a reason to call this in a constructor? I think this is prone to a race condition, i.e.: a filter added after this class had already initialized. I'd suggest for calling the filter when the value is to be used.

Reply:
You're right, I was also worried about having a race condition here, although there are not, there is no reason to keep it here, as we only use `allowed_promo_notes` inside `activate_promo_note`. So moved it there in 8c26e80. Thanks for raising it.